src/components/common/SafeLoadingError/index.tsx

```diff
@@ -1,20 +1,72 @@
-import type { ReactElement, ReactNode } from 'react'
+import { addUndeployedSafe } from '@/features/counterfactual/store/undeployedSafeSlice'
+import { isUndeployedSafe } from '@/features/counterfactual/utils'
+import { useHasFeature } from '@/hooks/useChains'
+import { useAppDispatch } from '@/store'
+import { showNotification } from '@/store/notificationsSlice'
+import { FEATURES } from '@/utils/chains'
+import { type ReactElement, type ReactNode, type ChangeEvent } from 'react'
 import { Button } from '@mui/material'
 import useSafeInfo from '@/hooks/useSafeInfo'
 import PagePlaceholder from '../PagePlaceholder'
 import { AppRoutes } from '@/config/routes'
 import Link from 'next/link'
 
+import css from './styles.module.css'
+
 const SafeLoadingError = ({ children }: { children: ReactNode }): ReactElement => {
   const { safeError } = useSafeInfo()
+  const dispatch = useAppDispatch()
+  const isCounterfactualEnabled = useHasFeature(FEATURES.COUNTERFACTUAL)
 
   if (!safeError) return <>{children}</>
 
+  const handleUpload = (e: ChangeEvent<HTMLInputElement>) => {
+    if (e.target.files) {
+      const file = e.target.files[0]
+      const reader = new FileReader()
+      reader.onload = (event) => {
+        if (!event.target) {
+          return
+        }
+        if (typeof event.target.result !== 'string') {
+          return
+        }
+
+        const parsedData = JSON.parse(event.target.result)
+        if (isUndeployedSafe(parsedData)) {
+          const chainId = parsedData.chainId
+          const address = parsedData.safeAddress
+          const safeProps = parsedData.safeProps
+
+          dispatch(addUndeployedSafe({ chainId, address, safeProps }))
+        } else {
+          dispatch(
+            showNotification({
+              variant: 'error',
+              groupKey: 'safe-import-error',
+              message: 'Could not recover safe account',
+            }),
+          )
+        }
+      }
+      reader.readAsText(file)
+    }
+  }
+
   return (
     <PagePlaceholder
       img={<img src="/images/common/error.png" alt="A vault with a red icon in the bottom right corner" />}
       text="This Safe Account couldn't be loaded"
     >
+      {isCounterfactualEnabled && (
+        <>
+          <Button component="label" variant="outlined" size="large" sx={{ mt: 2, mb: 1 }}>
+            Recover your safe setup
```

Comment:
How would the user even land on this page if they lost their safe address?

Reply:
This would only cover the case where the user knows their safe address and has a backup and they tried to open that safe on a new browser or different device.

If they lost their safe address but have a backup there is currently not an option to recover that backup from outside a safe route -> We could add this e.g. in the load safe flow

If they lost their safe address and don't have a backup the only thing they can do is go through the safe creation flow again to create "another" 1/1 safe and if nothing about the proxy version has changed it would result in the same address which they could back up then.

Reply:
If they have a Safe address isn't it trivial to infer the nonce/Safe version?

Reply:
When the user opens a counterfactual safe thats not in local storage and they are connected with a wallet we could predict the safe address at the same time and see if it matches i.e. owner is the connected EOA, threshold is 1 out of 1, safe version is latest version, salt nonce is 0 and if it matches we could load it into localStorage. If it doesn't match we could increase the salt nonce x times and check each time so it would essentially be the same thing we do in the safe creation flow just whenever we show a load safe error.

Only issue I see is that if they just open some safe, chances are low that their wallet will be connected so we wouldn't be able to predict an address there.

If they connect their wallet but don't have a safe open we could predict addresses but there wouldn't be anything to match so that would also not be helpful.